up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto

```diff
@@ -41,331 +41,300 @@ service uDiscovery {
   option (version_minor) = 0;
   option (id) = 1;
 
-  // uDiscovery Node Change Notification that sends the Update message
+
   option (notification_topic) = {
     id: 0x8000,
       name: "NodeChange",
       message: "Notification"
   };
 
-    
+  /************************** Write APIs ************************************/
 
-  // Used by any uProtocol application or service to find service instances
-  // locations. The passed UUri contains valid UEntity, UResource, and UAuthority information
-  // for a query.
-  // What is returned is a list of UUris that match the query. Possible return values in status are:
-  // 1. code.OK: Query lookup was successful
-  // 2. code.NOT_FOUND: No matching UUris were found
-  // 3. code.INVALID_ARGUMENT: The passed UUri is invalid
-  // 4. code.PERMISSION_DENIED: The caller does not have permission to perform the query
-  rpc LookupUri(v1.UUri) returns (LookupUriResponse) {
-    option (method_id) = 1;
-  }
+  /* 
+  UEs shall call RegisterUE API to add themselves to discovery.
+ */
 
-  // Update a node in the database.
-  // What is returned is the status of the API request as uprotocol.v1.UStatus.
-  // **NOTE:** You MUST have write permission to the node in the database
-  rpc UpdateNode(UpdateNodeRequest) returns (uprotocol.v1.UStatus) {
-    option (method_id) = 2;
-  }
+  /* Example: 
+  RegisterUE(“//ultifi.gm.com/core.example/2.0")
+  Response: Status: code.OK
+ */
 
-  // Query the database to find Node(s). What is passed is the search criterial in
-  // the FindNodeRequest message. the returned FindNodeResponse contains the
-  // resulting query node(s) and their child nodes, as well as the
-  // uprotocol.v1.UStatus for the query. Below are some example queries:
-  // 1. uDomain: `//*.device/`
-  // 2. uDevice: `//device`
-  // 3. uService: `//device.domain/body.access/`
-  // 4. uResource: `//device.domain/body.access//door.door_front_left`
-  // **NOTE:** You MUST have read permission to the node in the database
-  rpc FindNodes(FindNodesRequest) returns (FindNodesResponse) {
-    option (method_id) = 3;
+  rpc RegisterUE(RegisterUERequest) returns ( uprotocol.v1.UStatus){
+    option (method_id) = 1;
   }
 
-  // Query the database to fetch a list of 1 or more properties for a given node.
-  rpc FindNodeProperties(FindNodePropertiesRequest) returns (FindNodePropertiesResponse) {
-    option (method_id) = 4;
-  }
+    /* 
+  UEs shall call UnregisterUE API to remove themselves from discovery.
```

Comment:
What does registering and unregistering do? Is this something that a uE actually does? How do you ensure that we don't call the API to spoof being someone else?